src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -764,7 +764,7 @@ private function registerHttpCacheConfiguration(array $config, ContainerBuilder
         if ($httpMethodOverride) {
             $container->getDefinition('http_cache')
                   ->addArgument((new Definition('void'))
-                      ->setFactory([Request::class, 'enableHttpMethodParameterOverride'])
+                      ->setFactory(Request::enableHttpMethodParameterOverride(...))
```

Comment:
@nicolas-grekas Does the DI component supports dumping the container when using first-class callables for a factory ? I don't remember whether this has been added (and if yes, in which version), as the dumper does not sure arbitrary Closure.

but otherwise, I would be fine reverting that for all those `setFactory` calls. `setFactory` does not expect a callable. It expects a definition for producing a callable.

Reply:
AFAIU it doesn't support Closures so I'm reverting these changes. I'll open an issue to keep track of this potential improvement.